In [49]:
import pandas as pd
import numpy as np

#read data from data/data_all.csv

data_all = pd.read_csv('../data/data_all.csv')

In [50]:
data_all.describe()

index          Price      Rating   num_cores  num_threads  \
count   991.000000     991.000000  991.000000  991.000000   991.000000   
mean    505.860747   77266.504541   63.931382    8.128153    12.191726   
std     287.899458   57384.910269   10.190575    4.215499     5.585115   
min       1.000000    9800.000000   24.000000    2.000000     0.000000   
25%     258.500000   43595.000000   58.000000    6.000000     8.000000   
50%     507.000000   61900.000000   64.000000    8.000000    12.000000   
75%     754.500000   89245.000000   71.000000   10.000000    16.000000   
max    1002.000000  454490.000000   89.000000   24.000000    32.000000   

       ram_memory  primary_storage_capacity  secondary_storage_capacity  \
count  991.000000                991.000000                  991.000000   
mean    13.047427                610.938446                    4.004036   
std      5.591188                266.927666                   33.553936   
min      2.000000                 32.000000                    0.000000   
25%      8.000000                512.000000                    0.000000   
50%     16.000000                512.000000                    0.000000   
75%     16.000000                512.000000                    0.000000   
max     36.000000               2048.000000                  512.000000   

       display_size  resolution_width  resolution_height  
count    991.000000        991.000000         991.000000  
mean      15.171241       2003.503532        1181.227043  
std        0.938089        361.965292         263.884019  
min       10.100000       1080.000000         768.000000  
25%       14.000000       1920.000000        1080.000000  
50%       15.600000       1920.000000        1080.000000  
75%       15.600000       1920.000000        1200.000000  
max       18.000000       3840.000000        2560.000000

In [51]:
data_all.head()

index   brand                                              Model  Price  \
0      1   tecno  Tecno Megabook T1 Laptop (11th Gen Core i3/ 8G...  23990   
1      2   tecno  Tecno Megabook T1 Laptop (11th Gen Core i7/ 16...  35990   
2      3      hp  HP Victus 15-fb0157AX Gaming Laptop (AMD Ryzen...  51100   
3      4    acer  Acer Extensa EX214-53 Laptop (12th Gen Core i5...  39990   
4      5  lenovo  Lenovo V15 82KDA01BIH Laptop (AMD Ryzen 3 5300...  28580   

   Rating processor_brand processor_tier  num_cores  num_threads  ram_memory  \
0      63           intel        core i3          2            4           8   
1      67           intel        core i7          4            8          16   
2      73             amd        ryzen 5          6           12           8   
3      62           intel        core i5         12           16           8   
4      62             amd        ryzen 3          4            8           8   

   ... secondary_storage_type  secondary_storage_capacity gpu_brand  \
0  ...   No secondary storage                           0     intel   
1  ...   No secondary storage                           0     intel   
2  ...   No secondary storage                           0       amd   
3  ...   No secondary storage                           0     intel   
4  ...   No secondary storage                           0       amd   

     gpu_type is_touch_screen display_size  resolution_width  \
0  integrated           False         15.6              1920   
1  integrated           False         15.6              1920   
2   dedicated           False         15.6              1920   
3  integrated           False         14.0              1920   
4  integrated           False         15.6              1920   

   resolution_height       OS  year_of_warranty  
0               1080  windows                 1  
1               1080  windows                 1  
2               1080  windows                 1  
3               1080  windows                 1  
4               1080  windows                 1  

[5 rows x 22 columns]

In [52]:
data_all.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 991 entries, 0 to 990
Data columns (total 22 columns):
 #   Column                      Non-Null Count  Dtype  
---  ------                      --------------  -----  
 0   index                       991 non-null    int64  
 1   brand                       991 non-null    object 
 2   Model                       991 non-null    object 
 3   Price                       991 non-null    int64  
 4   Rating                      991 non-null    int64  
 5   processor_brand             991 non-null    object 
 6   processor_tier              991 non-null    object 
 7   num_cores                   991 non-null    int64  
 8   num_threads                 991 non-null    int64  
 9   ram_memory                  991 non-null    int64  
 10  primary_storage_type        991 non-null    object 
 11  primary_storage_capacity    991 non-null    int64  
 12  secondary_storage_type      991 non-null    object 
 13  secondary_storage_capacity  991 non

From this basic information we observe:
<ul> 
<li> our dataset has 991 record, but for EDA and clustering we will use only 80% of it, other 20% goes to our validation team</li>
<li> our dataset has 22 columns: 10 - int64, 1 - float64, 1 - bool, 10 - object (string)</li>
<li> columns, index and Model do not contain any useful information (almost all info, written in Model already exists in other columns)</li>
</ul>

<h1> DIVISION INTO VALIDATION SET(FOR VALIDATION TEAM) AND OUR DATASET <h1>

In [53]:
from sklearn.model_selection import train_test_split

data, validation_data = train_test_split(data_all, test_size=0.2, random_state=42)

validation_data.to_csv('../data/validation_data.csv', index=False)
data.to_csv('../data/our_data.csv', index=False)

<h1>Exploration of columns</h1>

In [54]:
print(data.columns)

Index(['index', 'brand', 'Model', 'Price', 'Rating', 'processor_brand',
       'processor_tier', 'num_cores', 'num_threads', 'ram_memory',
       'primary_storage_type', 'primary_storage_capacity',
       'secondary_storage_type', 'secondary_storage_capacity', 'gpu_brand',
       'gpu_type', 'is_touch_screen', 'display_size', 'resolution_width',
       'resolution_height', 'OS', 'year_of_warranty'],
      dtype='object')


In [60]:
pd.unique(data['brand'])

array(['acer', 'lenovo', 'dell', 'asus', 'msi', 'hp', 'samsung', 'jio',
       'apple', 'infinix', 'iball', 'lg', 'zebronics', 'tecno', 'fujitsu',
       'microsoft', 'chuwi', 'avita', 'ultimus', 'honor', 'gigabyte',
       'wings', 'realme', 'walker', 'primebook'], dtype=object)

column 'brand' has 25 different values

In [ ]:
# barplot of data['brand']



<h1>TRANSFORMATION OF OBJECT COLUMNS</h1>

There are no zeros in year_of_warranty, so probably "No information" means 0 years of warranty

In [38]:
def transform_column(year_of_warranty):
    if year_of_warranty == '1':
        return 1
    if year_of_warranty == '2':
        return 2
    if year_of_warranty == '3':
        return 3
    if year_of_warranty == 'No information':
        return 0

data['year_of_warranty'] = data['year_of_warranty'].map(transform_column)